runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -33,6 +38,19 @@
  */
 @Slf4j
 public class PersistedStreamingMap<K, V> implements ContextAwareMap<K, V> {
+    // Accessor/Mutator threads can interleave in a way that create a deadlock because they can create a
+    // circular dependency between the VersionLockedObject(VLO) lock and the common forkjoin thread pool. In order
+    // to break the dependency, parallel stream operations have to execute on a separate pool that applications
+    // cant use. For example, if there are 4 accessor threads all using the common forkjoin pool, one of the threads
+    // can acquire the VLO lock and cause the other 3 threads to wait, but after acquiring the VLO lock, the thread
+    // gets block on parallel stream, because the pool is exhausted with threads that are trying to acquire the VLO
+    // look, which creates a circular dependency. In other words, a deadlock.
+    private final static ForkJoinPool pool = new ForkJoinPool(Runtime.getRuntime().availableProcessors() - 1,
+            pool -> {
+                final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
+                worker.setName("PersistedStreamingMap-Forkjoin-pool-" + worker.getPoolIndex());
+                return worker;
+            }, null, true);
```

Comment:
`PersistedStreamingMap` cannot take advantage of parallel processing, so there is no point of spawning more than one thread. Is there a reason why `ForkJoinPool` was moved out of `CorfuTable`?

Reply:
I was not very clear about PersistedStreamingMap.

ForkJoinPool was moved out of CorfuTable because I need `StreamingMapDecorator.scanAndFilter` to get rid of fromMap. If PersistedStreamingMap cannot take advantage of parallel processing, I should put ForkJoinPool to StreamingMapDecorator.

Reply:
Just to give you a bit of context.

`PersistedStreamingMap` uses `RocksDB` underneath, which by design does not allow multi-threaded data iteration. So in this case, you can just scan the data using the same client thread -- no need for a new thread pool executor.

`StreamingMapDecorator` on the other hand has no such restrictions, and this is why `parallelStream()` can be used. However, there was a bug with using a common thread pool that would cause deadlock in some circumstances. This is why `ForkJoinPool` was introduced.

Reply:
Thanks for the context. I thought RocksDB iterator could somehow split data and operate in parallel so moved it here.

I have moved it to `StreamingMapDecorator`